File: Bpa_v2_front/src/app/models/facturecomp.ts
import { Etat } from "./etat";

export interface Facturecomp{
    idFacture: number;
    uid: string;
    mere: string;
    societe: string;
    fournisseur : string;
    objet : string;
    date: string;
    montant: string;
    devise: string;
    etat: Etat;
}

File: Bpa_v2_bakc/src/main/java/com/example/Bpa_v2_bakc/entities/sql_server/Facturecomp.java
package com.example.Bpa_v2_bakc.entities.sql_server;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "facturecomp")
public class Facturecomp {
    @Id 
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_facturecomp")
    private int idFacturecomp;

    private String uid;
    
    private String reffc;

    private String mere;
    private String societe;
    private String fournisseur;
    private String code;
    private String objet;
    private String date;
    private String montant;
    private String devise;
}
